### ISSUE TITLE
Peertube-Docker: Customize E-Mails (Subject, Templates, ..)

### ISSUE THREAD

**Issue by florianwgnr** [Chocobozzz/PeerTube#5673]
### Describe the problem to be solved

1. Currently, the Docker [custom-environment-variables.yaml](https://github.com/Chocobozzz/PeerTube/blob/master/support/docker/production/config/custom-environment-variables.yaml) does not contain variables for customizing the e-mail body-signature or subject prefix while it can be customized within the [production.yaml](https://github.com/Chocobozzz/PeerTube/blob/develop/config/production.yaml.example)
 2. emails are stored within the container and cannot be modified 
3. the email-subject is not stored within the email-templates

### Describe the solution you would like

In order to fully customize the email-templates the following steps should solve this issue:

1. add body-signature and subject-prefix to custom-environment-variables.yaml and make them usable in the .env-file
 
2. In order to modify the e-mail content, add a docker volume (e.g. docker-volume/emails) where [the folder containing the e-mail templates ](https://github.com/Chocobozzz/PeerTube/tree/develop/server/lib/emails) is mapped to the filesystem and can be customized
 
3. Move the e-mail subject from [emailer.ts](https://github.com/Chocobozzz/PeerTube/blob/develop/server/lib/emailer.ts) to the [the e-mail templates ](https://github.com/Chocobozzz/PeerTube/tree/develop/server/lib/emails)

Steps 3 might also be useful for #912

**Comment by cheako** [comment]
tldr; If I could know what this `yaml` file is supposed to be called(including where in the docker image it is currently) and a general idea of where(relative to `$CWD`) on the host system it should be found.  Then I may be able to submit a \*PR.  While I'm at it are there any other issues I should look into?

Think this issue demonstrates my issue nicely.  Wanted to configure Remote Storage and Redundancy, let me know if these are separate issues and if that's two or three issues total.

Is what needed to address these three issues to expose `production.yaml`, or is it `default.yaml`?  By expose, I mean there is a lack of documentation on how this file is supposed to be edited.  Are there more files that should be moved to a volume?  These are files that don't change version to version but do change deployment to deployment

\* To update the Dockerfile so a new location is exported via a volume or perhaps the fix is to move this file elsewhere that's already a volume?

My skills are no clue with nodejs or node-config and a handful of times wrote or edited a dockerfile.  Still willing to help.

**Comment by SnejPro** [comment]
> tldr; If I could know what this `yaml` file is supposed to be called(including where in the docker image it is currently) and a general idea of where(relative to `$CWD`) on the host system it should be found. Then I may be able to submit a *PR. While I'm at it are there any other issues I should look into?

I solved it by creating `production.yaml` (as copy of `default.yaml`) in `docker-volume/config/production.yaml`

**Comment by SnejPro** [comment]
> I solved it by creating `production.yaml` (as copy of `default.yaml`) in `docker-volume/config/production.yaml`

But i commented out everything, except the parameters i wanted to set, because i don't know if there could be conflicts with my .env file.

**Comment by cheako** [comment]
> But i commented out everything, except the parameters i wanted to set, because i don't know if there could be conflicts with my .env file.

I was thinking the same thing.  This feels ill-advised until this is better understood, I think the node-config documentation could be helpful here.

The other issue is the host/guest username mismatch, so a run-as uid/gid is advised.

**Comment by cheako** [comment]
This can be closed and I believe the problem is because so much of the config is suggested to be done in `.env`.  When in reality it should be clearly explained that `.env` and `docker-volume/config/production.yaml` are combined, use either.

**Comment by Chocobozzz** [comment]
Merging this issue in https://github.com/Chocobozzz/PeerTube/issues/6591
